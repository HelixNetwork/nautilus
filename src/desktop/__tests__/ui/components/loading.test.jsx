import React from 'react';
import { shallow } from 'enzyme';

import Loading from 'ui/components/loading';


const props = {
    loop : true,
    inline :true,
    transparent : true,
    onEnd : jest.fn(),
};

describe('Test for Loading component', () => {
    test('Render the Loading component', () => {
        const wrapper = shallow(<Loading {...props} />);

        expect(wrapper).toMatchSnapshot();
    });

    test('Checks the inline true state', () => {
        const wrapper = shallow(<Loading {...props} />);
        
        expect(wrapper.hasClass('inline')).toBeTruthy();
    });

    test('Checks the inline false state', () => {
        const mockProps = Object.assign({}, props, { inline: false });
        const wrapper = shallow(<Loading {...mockProps} />);

        expect(wrapper.hasClass('inline')).toBeFalsy ();
    });


    
});