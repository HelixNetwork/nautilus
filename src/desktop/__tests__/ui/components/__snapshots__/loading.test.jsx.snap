// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Loading component Render the Loading component 1`] = `
<div
  className="loading inline transparent"
>
  <Logo
    animate={true}
    loop={true}
    onEnd={[MockFunction]}
    size={300}
  />
</div>
`;
